inspirehep/modules/records/texkeys.py

```diff
@@ -0,0 +1,332 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Texkeys util functions."""
+
+from __future__ import absolute_import, division, print_function
+
+from datetime import date
+from datetime import datetime
+from itertools import chain
+import random
+import re
+import string
+import sys
+import unicodedata
+from unidecode import unidecode
+
+from invenio_db import db
+from invenio_pidstore.errors import PIDDoesNotExistError
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+from invenio_pidstore.providers.base import BaseProvider
+
+from inspirehep.utils.date import create_earliest_date
+from inspirehep.utils.helpers import force_list
+from inspirehep.utils.record import get_value
+
+IS_PYTHON_3 = sys.version_info[0] == 3
+
+
+def generate_texkey(record):
+    """It generates the texkey in according to the data inside the record.
+
+    :param record: the record that needs the texkey
+    :type record: dict
+
+    :return: return the new texkey if it is
+    able to produce it, otherwise empty string
+    :rtype: string
+    """
+    new_texkey = ''
+    first_part = first_part_generator(record)
+    if first_part:
+        new_texkey = first_part + ':' + second_part_generator(record) + third_part_generator()
+    return new_texkey
+
+
+def first_part_generator(record):
+    """It generates the first part of the texkey.
+
+    :param record: the record that needs the texkey
+    :type record: dict
+
+    :return: return the first part of the texkey
+    if it is able to produce it, otherwise empty string
+    :rtype: string
+    """
+    if 1 <= len(record.get('authors', [])) < 10:
+        first_part = sanitize(record.get('authors')[0]['full_name'])
+    elif record.get('collaborations', []):
+        first_part = sanitize(record.get('collaborations')[0]['value'])
+    elif record.get('corporate_author', []):
+        first_part = sanitize(record.get('corporate_author')[0])
+    elif 'proceedings' in record.get('document_type', []):
+        first_part = 'proceedings'
+    elif record.get('authors', []):
+        first_part = sanitize(record.get('authors')[0]['full_name'])
+    else:
+        first_part = ''
+
+    return first_part
+
+
+def second_part_generator(record):
```

Comment:
not in the record itself, apparently you can get it through [`record.created`](https://invenio-records.readthedocs.io/en/latest/api.html#invenio_records.api.RecordBase.created)